In [1]:
# 3.3 What Does This Code Do?
for row in range(10):
    for column in range(10):
        print('<' if row % 2 == 1 else '>', end='')
    print()

>>>>>>>>>>
<<<<<<<<<<
>>>>>>>>>>
<<<<<<<<<<
>>>>>>>>>>
<<<<<<<<<<
>>>>>>>>>>
<<<<<<<<<<
>>>>>>>>>>
<<<<<<<<<<


This code will print 10 '<' and a space in odd number row, print 10 '>' and a space in even number row and print 10 rows in total.

In [8]:
# 3.10 7% Investment Return
p = 1000
r = 0.07
for n in range(1, 31):
    a = p * (1 + r) ** n
    print("The amount you have is", round(a, 2), "after", n, "years")

The amount you have is 1070.0 after 1 years
The amount you have is 1144.9 after 2 years
The amount you have is 1225.04 after 3 years
The amount you have is 1310.8 after 4 years
The amount you have is 1402.55 after 5 years
The amount you have is 1500.73 after 6 years
The amount you have is 1605.78 after 7 years
The amount you have is 1718.19 after 8 years
The amount you have is 1838.46 after 9 years
The amount you have is 1967.15 after 10 years
The amount you have is 2104.85 after 11 years
The amount you have is 2252.19 after 12 years
The amount you have is 2409.85 after 13 years
The amount you have is 2578.53 after 14 years
The amount you have is 2759.03 after 15 years
The amount you have is 2952.16 after 16 years
The amount you have is 3158.82 after 17 years
The amount you have is 3379.93 after 18 years
The amount you have is 3616.53 after 19 years
The amount you have is 3869.68 after 20 years
The amount you have is 4140.56 after 21 years
The amount you have is 4430.4 after 22 years
T

In [3]:
#3.13 DS: Factorials
a = int(input('Enter an integer'))
n = 1
for i in range(1, a + 1):
    n *= i
print(n)

Enter an integer 4


24


In [ ]:
#3.31 DS: Categorical Data

Data like gender, city, and brand.

In [ ]:
#3.23 Validating Indentation

If we move the second print statement under the i in the if key-word, the third print statment wont work. However, tabnanny will not return an error.

In [32]:
#3.17 Nested Loops
#a)
for row in range(10):
    for column in range(int(row)+1):
        print('*', end='')
    print()
#b)
for row in reversed(range(10)):
    for column in range(int(row)+1):
        print('*', end='')
    print()
#c)
for row in range(10):
    for column in range(10):
        if column < row:
            print(' ', end='')
        else:
            print('*', end='')
    print()
#d)
for row in range(10):
    for column in range(10):
        if column < (9 - row):
            print(' ', end='')
        else:
            print('*', end='')
    print()

*
**
***
****
*****
******
*******
********
*********
**********
**********
*********
********
*******
******
*****
****
***
**
*
**********
 *********
  ********
   *******
    ******
     *****
      ****
       ***
        **
         *
         *
        **
       ***
      ****
     *****
    ******
   *******
  ********
 *********
**********


In [47]:
#Optinal EX
#3.14 Math Challenge: Approximating the Constant π
pie = 4
for i in range(1, 1000000):
    if i % 2 == 1:
        pie -= 4 / (2 * i + 1)
    else:
        pie += 4 / (2 * i + 1)
print(pie)

3.1415916535897743


About 600 terms we have 3.14.
About 1700 terms we have 3.141.
About 11000 terms we have 3.1415.
About 1000000 terms we have 3.14159.

In [68]:
#3.18 Challenge: Nested Looping
for row in range(10):
    for column in range(41):
        if column <= row:
            print('*', end='')
        elif column <= 10:
            print(' ', end='')
        elif column <= 20 - row:
            print('*', end='')
        elif column <= 20 + row:
            print(' ', end='')
        elif column <= 30:
            print('*', end='')
        elif column <= 39 - row:
            print(' ', end='')
        elif column <= 40:
            print('*', end='')
    print()

*          ********************         *
**         *********  *********        **
***        ********    ********       ***
****       *******      *******      ****
*****      ******        ******     *****
******     *****          *****    ******
*******    ****            ****   *******
********   ***              ***  ********
*********  **                ** *********
********** *                  ***********


In [ ]:
#3.24 Project: Using the prospector Static Code Analysis Tool

In [ ]:
#3.26 DS Research: Anscombe’s Quartet

I did all the required exercises and fully understand them.